torchx/lib/torchx.ex

```
@@ -228,7 +228,33 @@ defmodule Torchx do
   def eye(size, type, device), do: eye(size, size, type, device)
   defdevice eye(m, n, type, device)
   defdevice from_blob(blob, shape, type, device)
-  defdevice to_device(tensor, device)
+
+  @torch_function {:to_device, 2}
+  def to_device(tensor, device) do
+    {[tensor_ref], current_device} = prepare_tensors!([tensor])
+    {user_device, index} = normalize_device!(device)
+    target_device_struct = torch_device!(user_device, index)
+
+    tensor_to_move =
+      if user_device == :mps do
```

Agreed we should just let it raise on f64.